### ISSUE TITLE
Add file index to distinguish when files have same name 

### ISSUE THREAD

**Issue by tterletskyy** [siddhantgoel/streaming-form-data#138]
Hello,
`BaseTarget` class has two methods to set the next filename and content type from multipart/form-data when uploading multiple files.

```python
    def set_multipart_filename(self, filename: str):
        self.multipart_filename = filename

    def set_multipart_content_type(self, content_type: str):
        self.multipart_content_type = content_type
```

Could you please extend the logic with a new method that will increment the index of the file in the case when two or more files are uploaded with the same name like `test.txt` because right now I believe I cannot distinguish which chunk belongs to the next file.

Thank you.

**Comment by siddhantgoel** [comment]
I believe that's the case when multiple files are uploaded on the same input field (#113). I don't know if incrementing the index is enough here.

**Comment by siddhantgoel** [comment]
I'm not sure if I understood incrementing the index of the file. Which index are you referring to?

**Comment by tropicoo** [comment]
I have a case when I need to upload multiple files in one payload with `name=file` which may have the same name and encrypt them chunk by chunk. Let's say three `README.md` files. In your code the `self.multipart_filename` will always be the `README.md`. In this case I cannot distinguish when the first file will end and the next starts. I can simply track the count how many times the `set_multipart_filename` method was called and in this way catch when the first chunk of the next file is received. Maybe there is a better way of doing this or the idea is to implement some logic to handle this in more convenient way. 

## FastAPI code
```python
import logging

from fastapi import FastAPI, Request
from streaming_form_data import StreamingFormDataParser
from streaming_form_data.targets import BaseTarget

log = logging.getLogger(__name__)

app = FastAPI()


class PayloadTarget(BaseTarget):
    def on_data_received(self, chunk: bytes) -> None:
        log.critical(self.multipart_filename)
        log.critical(self.multipart_content_type)
        print(chunk)


class FileTarget(BaseTarget):
    def __init__(self, *args, **kwargs) -> None:
        super().__init__(*args, **kwargs)
        self._processing_file: int = 0

    def set_multipart_filename(self, filename: str):
        self._processing_file += 1
        log.critical('#' * 40)
        log.critical('Called with name "%s" %s time', filename, self._processing_file)
        log.critical('#' * 40)
        super().set_multipart_filename(filename)

    def on_data_received(self, chunk: bytes) -> None:
        log.critical('*' * 40)
        log.critical('Current filename: %s', self.multipart_filename)
        log.critical('Processing file #%d', self._processing_file)
        log.critical('*' * 40)
        


@app.post('/upload')
async def upload_file(request: Request) -> None:
    parser = StreamingFormDataParser(headers=request.headers)
    parser.register('file', FileTarget())
    parser.register('payload', PayloadTarget())
    async for chunk in request.stream():
        parser.data_received(chunk)
```

## Output:
```python
########################################
Called with name "README.md" 1 time
########################################
****************************************
Current filename: README.md
Processing file #1
****************************************
########################################
Called with name "README.md" 2 time
########################################
****************************************
Current filename: README.md
Processing file #2
****************************************
########################################
Called with name "README.md" 3 time
########################################
****************************************
Current filename: README.md
Processing file #3
****************************************
########################################
Called with name "2023-10-14-02.tsv.gz" 4 time
########################################
****************************************
Current filename: 2023-10-14-02.tsv.gz
Processing file #4
****************************************
None
application/json
b'{"name": "Taras"}'
```

## Postman:
![image](https://github.com/user-attachments/assets/af49f64b-57c8-413b-ad2c-be32eb2c1cdd)

**Comment by siddhantgoel** [comment]
This sounds like a duplicate of #113. I'm currently experimenting with an approach to allow multiple targets for the same field name which should handle what you described.

Could you post the raw request body (multipart/form-data encoded) that the server receives in such a case? It'll help with additional testing.

**Comment by siddhantgoel** [comment]
I just published v1.19.0 that includes a [`MultipleTargets`](https://github.com/siddhantgoel/streaming-form-data/blob/main/docs/index.md#multipletargets) target that might be helpful to you.
